### PR TITLE
Simplify Deprecated CMake Packages as Wrappers Around Current Packages

### PR DESCRIPTION
**Pull request by vector-of-bool**
The old CMake config-file-packages were generated by mostly-identical scripts and processes, but are deprecated in favor of our newer config-file-packages. I think it is safe at this point to replace the content of the deprecated package scripts with scripts that import the new packages and expose them in a mostly-compatible\* manner.

The legacy package generation significantly cluttered and complicated the CMakeLists for the respective projects, since a lot of work was done to populate variables that were inserted into the script templates, whereas now those usage requirements are propagated via targets alone.

The new deprecated scripts no longer define/use the `INCLUDE_DIRS`/`DEFINES`/etc variables, and only `XYZ_LIBRARIES`/`XYZ_LIBRARY` remain, which just contain the name of the equivalent imported target from the new CMake config-file-packages.

\* "mostly compatible" because the primary case of linking to e.g. `${BSON_LIBRARIES}` will "just work" with `target_link_libraries`, but users that depended on the specific `find_library` call and/or inspected other variables e.g. `BSON_INCLUDE_DIRS` will find that those are now not available. I think it is reasonable to assume that very few users, if any, were relying on this exact case.

(This PR contains commits that are already merged by #1382.)

### CODE REVIEW

File: src/libbson/CMakeLists.txt [hunk 1]
```diff
@@ -23,18 +33,11 @@ include (InstallRequiredSystemLibraries)
...
-
-message (STATUS "lib${BSON_OUTPUT_BASENAME} version (from VERSION_CURRENT file): ${BSON_VERSION}")
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 
 set (BSON_API_VERSION 1.0)
 
-set (CPACK_PACKAGE_VERSION_MAJOR ${BSON_MAJOR_VERSION})
-set (CPACK_PACKAGE_VERSION_MINOR ${BSON_MINOR_VERSION})
-
```
Comment (vector-of-bool) [inline src/libbson/CMakeLists.txt:37]: These values are inferred from `project()`, which now contains the correct version information.

(Not sure if we have many CPack users anyway, though)

File: src/libbson/CMakeLists.txt [hunk 2]
```diff
@@ -367,11 +367,6 @@ install (
    INCLUDES DESTINATION ${BSON_HEADER_INSTALL_DIR}
    FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
-set_target_properties(${TARGETS_TO_INSTALL} PROPERTIES
-   pkg_config_NAME libbson
-   pkg_config_DESCRIPTION "The libbson BSON serialization library."
-   pkg_config_VERSION "${BSON_VERSION}"
-)
```
Comment (vector-of-bool) [inline src/libbson/CMakeLists.txt:374]: Inferred from `project()`

File: src/libbson/CMakeLists.txt [hunk 3]
```diff
@@ -390,22 +385,7 @@ if (ENABLE_APPLE_FRAMEWORK)
...
-   endif ()
-endforeach ()
-# System dependencies don't match the above regexs, but also don't want the -l
-foreach (_lib ${BSON_SYSTEM_LIBRARIES})
-   set (LIBBSON_LIBRARIES "${LIBBSON_LIBRARIES} ${_lib}")
-endforeach ()
-
 include (CMakePackageConfigHelpers)
-set (INCLUDE_INSTALL_DIRS "${BSON_HEADER_INSTALL_DIR}")
-set (LIBRARY_INSTALL_DIRS ${CMAKE_INSTALL_LIBDIR})
```
Comment (vector-of-bool) [inline src/libbson/CMakeLists.txt:408]: Only used by the old script generation during `configure_file`, so these aren't needed otherwise.

**Review by rcsanchez97 — APPROVED** [review]
Nice! I'm all about simplifying and eliminating unnecessary code. Just a few typo fix recommendations, but aside from that LGTM.

File: build/cmake/libmongoc-1.0-config.cmake.in [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+endif()
+
+set (MONGOC_MAJOR_VERSION @libmongoc_VERSION_MAJOR@)
+set (MONGOC_MINOR_VERSION @libmongoc_VERSION_MINOR@)
+set (MONGOC_MICRO_VERSION @libmongoc_VERSION_PATCH@)
+set (MONGOC_VERSION @libmongoc_VERSION@)
+set (MONGOC_VERSION_FULL @libmongoc_VERSION_FULL@)
+
+include(CMakeFindDependencyMacro)
+find_dependency (mongoc-1.0 "@libmongoc_VERSION_MAJOR@.@libmongoc_VERSION_MINOR@")
```
Comment (kevinAlbs) [inline build/cmake/libmongoc-1.0-config.cmake.in:26]: Is `@libmongoc_VERSION_MAJOR@.@libmongoc_VERSION_MINOR@` needed? libbson-1.0-config.cmake.in does does not include the version in the call to `find_dependency`.

**Pull request merged**